olcPixelGameEngine.h

```diff
@@ -2332,6 +2336,105 @@ namespace olc
 		}
 	}
 
+
+  void PixelGameEngine::DrawPolygon(const std::vector<olc::vi2d>& points, olc::Pixel p)
+  {
+    // Maybe nothing to do?
+    if (points.size() <= 0) 
+      return;
+
+    // Draw every line and close the polygon
+    for (int i=0; i < points.size(); i++)
+    {
+      int32_t x1 = (int32_t)points[i].x;
+      int32_t y1 = (int32_t)points[i].y;
+      int32_t x2 = (int32_t)points[(i+1) % points.size()].x;
+      int32_t y2 = (int32_t)points[(i+1) % points.size()].y;
+      DrawLine(x1, y1, x2, y2, p);
+    }
+  }  
+
+  void PixelGameEngine::FillPolygon(const std::vector<olc::vi2d>& points, olc::Pixel p)
+  {
+    // Maybe nothing to do?
+    if (points.size() <= 0) 
+      return;
+
+    // Find the min and max Y-values
+    float starty = (float)vScreenSize.y;
+    float stopy = 0.f;
+    for (auto p: points)
+    {
+      if (p.y < starty) starty = p.y;
+      if (p.y > stopy)  stopy  = p.y;
+    }

```

Comment:
Just quickly scanned through and noticed a small optimisation you could make, which would be to iterate over references rather than copies of the elements of the std::vector of points.

for (auto& p: points)
{
  if (p.y < starty) starty = p.y;
  if (p.y > stopy)  stopy  = p.y;
}

Reply:
You are right! Thanks to notice.

Reply:
That's actually slower, copying such a small datastructure is much faster than dereferencing it from a reference.

Reply:
> That's actually slower, copying such a small datastructure is much faster than dereferencing it from a reference.

And I bet the compiler would notice only the y coordinate has to be copied and optimize the loop accordingly.